app/src/main/java/org/mifos/mobile/viewModels/RegistrationViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package org.mifos.mobile.viewModels
+
+import android.util.Patterns
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.observers.DisposableObserver
+import io.reactivex.schedulers.Schedulers
+import okhttp3.ResponseBody
+import org.mifos.mobile.api.DataManager
+import org.mifos.mobile.models.register.RegisterPayload
+import javax.inject.Inject
+
+class RegistrationViewModel @Inject constructor(var dataManager : DataManager?) : ViewModel() {
+    private val compositeDisposables: CompositeDisposable = CompositeDisposable()
+
+    private var registrationResultSuccess = MutableLiveData<Boolean>()
+    val readRegistrationResultSuccess : LiveData<Boolean> get() = registrationResultSuccess
+    private var exceptionOnRegistration : Throwable? = null
+    val readExceptionOnRegistration : Throwable get() = exceptionOnRegistration!!
+
+    fun isInputFieldBlank(fieldText : String) : Boolean {
+        return fieldText.trim { it <= ' '}.isEmpty()
+    }
+
+    fun isInputLengthInadequate(fieldText : String) : Boolean {
+        return fieldText.trim { it <= ' '}.length < 6
+    }
+
+    fun inputHasSpaces(fieldText: String) : Boolean {
+        return fieldText.trim { it <= ' '}.contains(" ")
+    }
+
+    fun hasLeadingTrailingSpaces(fieldText: String) : Boolean {
+        return fieldText.trim { it <= ' ' }.length < fieldText.length
+    }
+
+    fun isEmailInvalid(emailText : String) : Boolean {
+        return !Patterns.EMAIL_ADDRESS.matcher(emailText.trim { it <= ' ' }).matches()
+    }
+
+    fun registerUser(registerPayload: RegisterPayload?) {
+        dataManager?.registerUser(registerPayload)
```

Comment:
Coroutines, please!

Reply:
@jawidMuhammadi am I missing something, we were just performing MVVM migration right now?

Reply:
At some point, we'll need to migrate it. It was on our idea list and roadmap, and I hope you'll be able to implement it by the end of GSoC. I would have used the coroutine which is much easier for me. Howser, still it is all up to you guys on how to move forward and implement them.

Reply:
That's a great point Jawid. I'm up for it. I'll replace RxJava calls with coroutines. Great!!

Reply:
Hey @jawidMuhammadi , I've made the changes except this one. Because I'm right now researching and learning how to use properly use coroutines in our project. Please have a look at this whenever you find time. Plus right now I didn't put created repository in a new Package because that would conflict with other contributor's PRs as well. For creating packages we can do a whole new PR.